crates/primitives/transactions/src/lib.rs

```diff
@@ -190,3 +191,17 @@ pub struct HandleL1MessageTransaction {
     pub entry_point_selector: Felt252Wrapper,
     pub calldata: Vec<Felt252Wrapper>,
 }
+
+impl TryFrom<MsgFromL1> for HandleL1MessageTransaction {
+    type Error = Error;
+
+    fn try_from(msg: MsgFromL1) -> Result<Self, Self::Error> {
+        let calldata = msg.payload.into_iter().map(|felt| felt.into()).collect();
+        Ok(Self {
+            contract_address: msg.to_address.into(),
+            nonce: 0u32.into(),
```

Comment:
I don't think you should do that. They need a nonce.
Try `impl From<(Nonce, MsgFromL1)>` instead.

Or don't impl from. If you are just using it once in you very special case where you want nonce to be 0.

Reply:
If you send a real message from l1 with nonce 0 and then try to execute that, it will probably fail.

I think you have to disable nonce validation here: `&RuntimeExecutionConfigBuilder::new::<T>().with_query_mode().build(),`